### PR TITLE
Fix Kots existing cluster snapshot onboarding flow for LVP plugin

### PR DESCRIPTION
**Pull request by sgalsaleh**
#### What this PR does / why we need it:

Fixes an issue where existing cluster snapshot onboarding flow does not work when using the local volume provider plugin.

#### Which issue(s) this PR fixes:

Fixes [SC-46486](https://app.shortcut.com/replicated/story/46486/kots-existing-cluster-snapshot-onboarding-flow-for-hostpath-or-nfs-snapshots-without-minio-is-broken)

#### Does this PR introduce a user-facing change?

```release-note
Fixes an issue where existing cluster snapshot onboarding flow does not work when using the local volume provider plugin.
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: pkg/snapshot/filesystem_lvp.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+				NFS: &types.NFSConfig{
+					Path:   bsl.Spec.Config["path"],
+					Server: bsl.Spec.Config["server"],
+				},
+			}
+			return fileSystemConfig, nil
+		}
+	}
+
+	// backup storage location does not exist, get file system config from the config map
```
Comment (sgalsaleh) [inline pkg/snapshot/filesystem_lvp.go:71]: this is new

File: pkg/snapshot/filesystem_lvp.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+// one customizes the LVP plugin deployment with a config map based on the chosen file system backing and the detection of Openshift.
+// This ensures the Velero and Restic have permissions to write to the disk.
+// the second config map is used to store the current file system configuration.
+func EnsureLocalVolumeProviderConfigMaps(deployOptions FileSystemDeployOptions, veleroNamespace string) error {
+	clientset, err := k8sutil.GetClientset()
+	if err != nil {
+		return errors.Wrap(err, "failed to get kubernetes clientset")
+	}
+
+	if err := ensureFileSystemLVPConfigMap(context.TODO(), clientset, deployOptions); err != nil {
```
Comment (sgalsaleh) [inline pkg/snapshot/filesystem_lvp.go:153]: this is new

File: pkg/snapshot/filesystem_lvp.go [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+
+	_, err = clientset.CoreV1().ConfigMaps(deployOptions.Namespace).Update(ctx, existingConfigMap, metav1.UpdateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to update deployment")
+	}
+
+	return nil
+}
+
+func fileSystemLVPConfigMapResource(fileSystemConfig types.FileSystemConfig) *corev1.ConfigMap {
```
Comment (sgalsaleh) [inline pkg/snapshot/filesystem_lvp.go:252]: this is new

File: pkg/snapshot/filesystem_lvp.go [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+
+// RevertToMinioFS will apply the spec of the previous BSL to the current one and then update.
+// Used for recovery during a failed migration from Minio to LVP.
+func RevertToMinioFS(ctx context.Context, kotsadmNamespace, veleroNamespace string, previousBsl *velerov1api.BackupStorageLocation) error {
+	bsl, err := FindBackupStoreLocation(context.TODO(), kotsadmNamespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to find backupstoragelocations")
+	}
+
+	bsl.Spec = previousBsl.Spec
```
Comment (divolgin) [inline pkg/snapshot/filesystem_lvp.go:111]: bsl can be nil

File: pkg/snapshot/filesystem_lvp.go [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+		case SnapshotStoreNFSProvider:
+			fileSystemConfig := &types.FileSystemConfig{
+				NFS: &types.NFSConfig{
+					Path:   bsl.Spec.Config["path"],
+					Server: bsl.Spec.Config["server"],
+				},
+			}
+			return fileSystemConfig, nil
+		}
+	}
```
Comment (divolgin) [inline pkg/snapshot/filesystem_lvp.go:69]: Should there be a return here if bsl is not nil but neither of the cases match?

File: pkg/snapshot/filesystem_lvp.go [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+	// backup storage location does not exist, get file system config from the config map
+	clientset, err := k8sutil.GetClientset()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get k8s clientset")
+	}
+
+	fileSystemConfigMap, err := clientset.CoreV1().ConfigMaps(namespace).Get(ctx, FileSystemLVPConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		if kuberneteserrors.IsNotFound(err) {
+			return nil, nil
```
Comment (divolgin) [inline pkg/snapshot/filesystem_lvp.go:80]: this is not new, but also some callers assume result is never nil

File: pkg/snapshot/filesystem_lvp.go (hunk 5 above)
Comment (sgalsaleh) [reply pkg/snapshot/filesystem_lvp.go:69]: no, cuz we wanna try getting the config from the configmap

Comment (sgalsaleh) [reply pkg/snapshot/filesystem_lvp.go:69]: hmm, maybe actually, idk. i'll think about it

**Review by divolgin — APPROVED** [review]

**Pull request merged**